src/main/java/fi/metatavu/metaform/server/rest/RealmsApiImpl.java

```diff
@@ -497,6 +508,53 @@ public Response updateEmailNotification(String realmId, UUID metaformId, UUID em
     return createOk(emailNotificationTranslator.translateEmailNotification(emailNotification));
   }
 
+  /**
+   * Resolves reply user from payload. If user has appropriate permissions user can 
+   * be other than logged user, otherwise logged user is returned
+   * 
+   * @param reply reply
+   * @return reply user id
+   */
+  private UUID getReplyUserId(Reply reply) {
+    UUID userId = reply.getUserId();
+    if (!isRealmMetaformAdmin() || userId == null) {
+      return getLoggerUserId();
+    }
+    
+    return userId;
+  }
+//  
+//  /**
+//   * Returns list of attachment ids from a reply
+//   * 
+//   * @param metaformEntity Metaform REST entity
+//   * @param data reply data
+//   * @return list of attachment ids from a reply
+//   */
+//  private List<UUID> getReplyAttachmentIds(Metaform metaformEntity, ReplyData data) {
+//    return fieldController.getFieldNamesByType(metaformEntity, MetaformFieldType.FILES).stream()
+//      .map((fieldName) -> {
+//        return StringUtils.split(String.valueOf(data.get(fieldName)));
+//      })
+//      .flatMap(Arrays::stream)
+//      .map(UUID::fromString)
+//      .collect(Collectors.toList());
+//  }
+//  
+//  /**
+//   * Persists previously uploaded files as attachments.
+//   * 
+//   * @param metaformEntity Metaform REST entity
+//   * @param data reply data
+//   * @param userId user id
+//   */
+//  private void persistReplyAttachments(Metaform metaformEntity, ReplyData data, UUID userId) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.metaform-api%3Ametaform-api&pullRequest=47&issues=AWVk3wFz4Bfu85TPTV1D&open=AWVk3wFz4Bfu85TPTV1D)

[ISSUE_KEY]: AWVk3wFz4Bfu85TPTV1D